lib/temporal/workflow.rb

```diff
@@ -19,6 +19,8 @@ def self.execute_in_context(context, input)
       Temporal.logger.error("Workflow execution failed with: #{error.inspect}")
       Temporal.logger.debug(error.backtrace.join("\n"))
 
+      Temporal::ErrorHandler.handle(error, context.send(:metadata))
```

Comment:
We should probably make `metadata` public, `.send` suggests that we're doing something wrong here

Reply:
Done.